lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -2612,8 +2612,8 @@ bool SwiftLanguageRuntimeImpl::GetDynamicTypeAndAddress(
 
     Flags subst_type_info(bound_type.GetTypeInfo());
     if (subst_type_info.AnySet(eTypeIsClass)) {
-      success = GetDynamicTypeAndAddress_Class(in_value, use_dynamic,
-                                               class_type_or_name, address);
+      success = GetDynamicTypeAndAddress_Class(
```

Comment:
Do we need to check that `bound_type != val_type` to prevent infinite loops, or does BindGenericTypeParameters() return {} when it didn't do anything?

Reply:
This isn't a recursive function, we're calling `GetDynamicTypeAndAddress_Class` from `GetDynamicTypeAndAddress`. I decided to do it this way to keep the same pattern as `GetDynamicTypeAndAddress_Protocol` and `GetDynamicTypeAndAddress_Value`.